cpp/tests/utilities/column_utilities.cu

```diff
@@ -141,9 +141,10 @@ void expect_equal_buffers(void const* lhs, void const* rhs,
 std::vector<bitmask_type> bitmask_to_host(cudf::column_view const& c) {
   if (c.nullable()) {
     auto num_bitmasks = bitmask_allocation_size_bytes(c.size()) / sizeof(bitmask_type);
+    auto mask = copy_bitmask(c.null_mask(), c.offset(), c.offset()+c.size());
```

Comment:
Just one optimization: 
if `c.offset()` is zero, you can directly use cudaMemCpy with `c.null_mask()`
```cpp
auto mask = (c.offset()==0 ? c.null_mask() : 
copy_bitmask(c.null_mask(), c.offset(), c.offset()+c.size()));
```

Reply:
The type differs, copy_bitmask returns `rmm::device_buffer` and null_mask returns `cudf::bitmask_type *`. If I just want to access data from `rmm::device_buffer`, I have to create an object from copy_bitmask which defeats the change we are trying to make.

Reply:
Type mismatch can be addressed. 
```cpp
rmm::device_buffer mask;
auto dev_bitmask; //use this for cudaMemcpy
if (c.offset()==0) {
   dev_bitmask=c.null_mask();
} else {
      mask = copy_bitmask(c.null_mask(), c.offset(), c.offset()+c.size()));
      dev_bitmask = mask.data(); 
} 
```